src/system.h

```diff
@@ -783,19 +783,54 @@ extern void SyAbortBags(const Char * msg) NORETURN;
 /****************************************************************************
 **
 *F * * * * * * * * * * * * * loading of modules * * * * * * * * * * * * * * *
+**
+** GAP_KERNEL_API_VERSION gives the version of the GAP kernel. This value
+** is used to check if kernel modules were built with a compatible kernel.
+** This version is not the same as, and not connected to, the GAP version.
+**
+** The kernel will not compile a version is stored similarly to 'semver',
+** as major*1000+minor.
+**
+** The algorithm used is the following:
+**
+** The kernel will not load a module compiled for a newer kernel.
+**
+** The kernel will not load a module compiled for a different major version.
+**
+** The minor version should be incremented when new backwards-compatible
+** functionality is added. The major version should be incremented when
+** a backwards-incompatible change is made.
+**
 */
 
+enum { GAP_KERNEL_API_VERSION = 1001 };
```

Comment:
So, how about splitting this, to avoid accidents and enforce the right format? Like so:
```C
enum {
    GAP_KERNEL_API_MAJOR_VERSION = 1,
    GAP_KERNEL_API_MINOR_VERSION = 1,
    GAP_KERNEL_API_VERSION = GAP_KERNEL_API_MAJOR_VERSION * 1000 + GAP_KERNEL_API_MINOR_VERSION;
};
```